pytorch_lightning/trainer/callback_hook.py

```diff
@@ -96,7 +96,15 @@ def on_train_epoch_end(self, outputs: EPOCH_OUTPUT):
             outputs: List of outputs on each ``train`` epoch
         """
         for callback in self.callbacks:
-            callback.on_train_epoch_end(self, self.lightning_module, outputs)
+            if is_param_in_hook_signature(callback.on_train_epoch_end, "outputs"):
+                warning_cache.warn(
+                    "`Callback.on_train_epoch_end` signature has changed in v1.3."
```

Comment:
```suggestion
                    "The signature of `Callback.on_train_epoch_end` has changed in v1.3."
```

Reply:
haha my copy/paste from @kaushikb11 's change in #6120 . will clean this up